src/retrie/trie.py

```diff
@@ -35,20 +52,81 @@ class Trie:
 
     __slots__ = "data"
 
-    def __init__(self):
+    def __init__(
+        self, *word  # type: Text
+    ):
         """Initialize data dictionary."""
         self.data = {}  # type: data_type
+        self.add(*word)
+
+    def __eq__(
+        self, other  # type: Any
+    ):  # type: (...) -> bool
+        """Compare two Trie objects."""
+        if not isinstance(other, Trie):
+            return False
+        return self.data == other.data
```

Comment:
These lines can fold into a single statement (and account for the subclassing case):
```
return self.__class__ == other.__class__ and self.data == other.data
```